### PR TITLE
SNOW-1332628: Fix and test for permission verification

### PR DESCRIPTION
**Pull request by sfc-gh-pmotacki**
### Description
SNOW-1332628:  Fix and test for permission verification

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: test/unit/file_transfer_agent/file_util_test.js [hunk 1]
```diff
@@ -50,3 +50,52 @@ describe('matching files by wildcard', function () {
...
+    testFilePath = await testUtil.createTempFileAsync(os.tmpdir(), testUtil.createRandomFileName());
+  });
+
+  after(async function () {
+    await fsPromises.rm(testFilePath);
+  });
+
+  [
+    {
+      permission: '600',
```
Comment (sfc-gh-dprzybysz) [inline test/unit/file_transfer_agent/file_util_test.js:68]: in code we change the suffix only - let's add the case with '600' prefixed with sth

**Review by sfc-gh-mhofman — APPROVED** [review]

File: lib/file_transfer_agent/file_util.js [hunk 2]
```diff
@@ -148,9 +148,9 @@ function getMatchingFilePaths(dir, fileName) {
 function validateOnlyUserReadWritePermission(filePath) {
   fs.accessSync(filePath, fs.constants.F_OK);
   const mode = (fs.statSync(filePath)).mode;
-  const permission = mode & 0o600;
+  const permission = mode | 0o600;
   //This should be 600 permission, which means the file permission has not been changed by others.
-  if (permission.toString(8) === '600') {
+  if (permission.toString(8).endsWith('600')) {
```
Comment (sfc-gh-fochnik) [inline lib/file_transfer_agent/file_util.js:153]: Why are we not comparing numerically to 0o600?

**Review by sfc-gh-dheyman-1 — APPROVED** [review]

File: test/unit/file_transfer_agent/file_util_test.js [hunk 3]
```diff
@@ -50,3 +50,56 @@ describe('matching files by wildcard', function () {
   });
 
 });
+
+
+describe('varify only user read/write permission', function () {
```
Comment (sfc-gh-pfus) [inline test/unit/file_transfer_agent/file_util_test.js:55]: typo - verify

File: test/unit/file_transfer_agent/file_util_test.js [hunk 4]
```diff
@@ -50,3 +50,56 @@ describe('matching files by wildcard', function () {
...
+  });
+
+  after(async function () {
+    await fsPromises.rm(testFilePath);
+  });
+
+  [
+    {
+      permission: '600',
+      testResult: true
```
Comment (sfc-gh-pfus) [inline test/unit/file_transfer_agent/file_util_test.js:69]: expected result?

**Review by sfc-gh-pfus — APPROVED** [review]

**Review by sfc-gh-fochnik — APPROVED** [review]

**Pull request merged**